core/io/logger.cpp

```diff
@@ -129,7 +130,7 @@ void RotatedFileLogger::clear_old_backups() {
 
 	da->list_dir_begin();
 	String f = da->get_next();
-	HashSet<String> backups;
+	RBSet<String> backups;
```

Comment:
I'd suggest adding a short comment to clarify what RBSet specifically is needed here, so this doesn't get refactored back by mistake.